### PR TITLE
Send events to Matomo when viewing pages

### PR DESCRIPTION
**Pull request by ptbrowne**
To help us understand better the behavior of users (only
for those who consented), page events are sent (on a new
page, or for a modal) to matomo.

### CODE REVIEW

File: src/ducks/transactions/actions/ReimbursementStatusAction/ReimbursementStatusModal.jsx [hunk 1]
```diff
@@ -13,91 +16,74 @@ import { isHealthExpense } from 'ducks/categories/helpers'
...
-      onChange,
-      t,
-      className,
-      breakpoints: { isMobile },
-      brands,
-      ...rest
-    } = this.props
-    const choices = ['pending', 'reimbursed', 'no-reimbursement']
-    const status = getReimbursementStatus(transaction)
+const ReimbursementStatusModal = props => {
```
Comment (Crash--) [inline src/ducks/transactions/actions/ReimbursementStatusAction/ReimbursementStatusModal.jsx:22]: to keep the same behavior you should have been using React.memo, no?

File: src/ducks/tracking/browser.jsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
+import React, { useContext, createContext } from 'react'
+import memoize from 'lodash/memoize'
+import {
+  getTracker as uiGetTracker,
+  shouldEnableTracking
```
Comment (Crash--) [inline src/ducks/tracking/browser.jsx:7]: I don't see it used?

File: src/ducks/transactions/actions/ReimbursementStatusAction/ReimbursementStatusModal.jsx [hunk 3]
```diff
@@ -26,6 +29,19 @@ const ReimbursementStatusModal = props => {
...
+  const tracker = useTracker()
+  const { categoryName, subcategoryName } = useParams()
+
+  useEffect(() => {
+    if (categoryName && subcategoryName) {
+      tracker.trackPage(
+        `analyse:${categoryName}:depense-remboursement`
+      )
+    } else {
+      tracker.trackPage(`mon_compte:depense:remboursement`)
```
Comment (Crash--) [inline src/ducks/transactions/actions/ReimbursementStatusAction/ReimbursementStatusModal.jsx]: wanted to have it in french?

File: src/ducks/transfers/steps/TransferState.jsx [hunk 4]
```diff
@@ -40,12 +42,21 @@ export const TransferSuccess = React.memo(function TransferSuccess({ onExit }) {
   )
 })
 
+const defaultErrorPageName = 'impossible'
+const konnectorErrorToPageName = {
+  LOGIN_FAILED: 'echec',
+  'VENDOR_DOWN.BANK_DOWN': 'banque-indisponible'
```
Comment (Crash--) [inline src/ducks/transfers/steps/TransferState.jsx:48]: C'est vraiment la string qu'on veut mapper ?

File: src/ducks/tracking/browser.jsx [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    }
+  }
+})
+
+export const getTracker = getMatomoTracker
+
+export const TrackerContext = createContext()
+
+export const TrackerProvider = ({ children }) => {
+  const tracker = getTracker()
```
Comment (Crash--) [inline src/ducks/tracking/browser.jsx:61]: Did you test with an account that disabled tracking? I think that tracker will be empty or undefined

**Review by Crash-- — APPROVED** [review]

File: src/ducks/transactions/actions/ReimbursementStatusAction/ReimbursementStatusModal.jsx (hunk 1 above)
Comment (ptbrowne) [reply src/ducks/transactions/actions/ReimbursementStatusAction/ReimbursementStatusModal.jsx:22]: Done

File: src/ducks/transfers/steps/TransferState.jsx (hunk 4 above)
Comment (ptbrowne) [reply src/ducks/transfers/steps/TransferState.jsx:48]: Oui ?

File: src/ducks/tracking/browser.jsx (hunk 2 above)
Comment (ptbrowne) [reply src/ducks/tracking/browser.jsx:7]: Fixed afterwards.

File: src/ducks/transactions/actions/ReimbursementStatusAction/ReimbursementStatusModal.jsx (hunk 3 above)
Comment (ptbrowne) [reply src/ducks/transactions/actions/ReimbursementStatusAction/ReimbursementStatusModal.jsx]: Oui

File: src/ducks/tracking/browser.jsx (hunk 5 above)
Comment (ptbrowne) [reply src/ducks/tracking/browser.jsx:61]: We return a shimmed tracker, methods are there but do not do anything. This way, components do not care about that.

**Comment by ptbrowne** [comment]
CI was red due to a build problem (cannot recognize JSX syntax) due to an old version of babel-preset-cozy-app.

**Pull request merged**